src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -96,22 +101,34 @@ public StudyService(
         this.webClient =  webClientBuilder.build();
 
         this.studyRepository = studyRepository;
+        this.studyBySubjectRepository = studyBySubjectRepository;
+        this.publicStudyRepository = publicStudyRepository;
     }
 
-    Flux<StudyInfos> getStudyList() {
-        Flux<Study> studyList = studyRepository.findAll();
-        return studyList.map(study -> new StudyInfos(study.getName(), study.getDescription(), study.getCaseFormat()));
+    Flux<StudyInfos> getStudyList(String userId) {
+        Flux<StudyBySubject> studyBySubjectFlux = studyBySubjectRepository.findAllByUserIdAndIsPrivate(userId, true);
+
+        return Flux.concat(studyBySubjectFlux.map(studyBySubject ->
+            new StudyInfos(studyBySubject.getStudyName(), studyBySubject.getUserId(), studyBySubject.getDescription(), studyBySubject.getCaseFormat())
+        ), getPublicStudyList());
+    }
+
+    Flux<StudyInfos> getPublicStudyList() {
+        Flux<PublicStudy> publicStudyFlux = publicStudyRepository.findAll();
+        return publicStudyFlux.map(publicStudy ->
+            new StudyInfos(publicStudy.getStudyName(), publicStudy.getUserId(), publicStudy.getDescription(), publicStudy.getCaseFormat())
+        );
     }
 
-    Mono<Study> createStudy(String studyName, UUID caseUuid, String description) {
+    @Transactional
```

Comment:
not needed since createStudy is @Transactional ?

Reply:
Ah I meant insertStudy. But I think you need this.insertStudy to use the proxy ?

Reply:
Nah, foo() and this.foo() are the same... What you did is best, ignore my remark